### DISCUSSION TITLE
Metals acting more like IntelliJ when it used to act like Eclipse

### DISCUSSION THREAD

**Discussion by ctataryn** [scalameta/metals#2423] [General]
One thing I had always admired about programming in the Eclipse IDE was the fact that it detected problems and updated it's auto-complete immediately after I changed something in one of my source files. Whereas in IntelliJ, I could change a file and then another editor window that had source code that is dependant wouldn't really pickup on the change until I did a full re-compile.

I loved the fact that Metals used to work like Eclipse, but I feel as though in more recent updates to the plugin it is acting more like IntelliJ.

Am I imagining this? Perhaps it has more to do with bloop than metals? Or a misconfiguration on my system?

**Comment by tgodzik** [comment]
So Metals always required a recompilation of sources to see the changes, Bloop only minimized the time for the recompilation. There is a long standing issue that is related and has a discussion about this topic https://github.com/scalameta/metals/issues/917

I would love to experiment with it when working with Scala 3 and maybe port the approach back to Scala 2, which should fix the issues you are seeing.

**Reply by ctataryn** [reply]
@tgodzik Thanks for the context. I'm overall really happy with the Metals extension, you folks have done a phenomenal job. One other thing I'll point out is that updates to build.sbt files don't always trigger the "Import build" popup. Sometimes I have to use the sidebar's "Import build" option manually, not sure if others have had this experience or not. I'm dealing with a multi-module sbt project if that makes a difference.

**Reply by tgodzik** [reply]
Hmm... this might be an issue, I think I've seen it, but always lacked time to investigate. We might need to fine tune the change detection better. Please don't hesitate to report an issue.
